src/solver/algorithm/strong/puzzle.rs

```diff
@@ -106,13 +244,576 @@ where
         .context("Failed to select solution set database table.")?;
 
     Ok(db)
+
+}
+*/
+
+// THIS IS ONLY FOR TESTING PURPOSES
+struct TestDB {
+    memory: HashMap<State, BitVec<u8, Msb0>>,
 }
 
+impl TestDB {
+    fn initialize() -> Self {
+        Self {
+            memory: HashMap::new(),
+        }
+    }
+}
+
+impl KVStore for TestDB {
+    fn put<R: crate::database::Record>(&mut self, key: State, record: &R) {
+        let new = BitVec::from(record.raw()).clone();
+        self.memory.insert(key, new);
+    }
+
+    fn get(
+        &self,
+        key: State,
+    ) -> Option<&bitvec::prelude::BitSlice<u8, bitvec::prelude::Msb0>> {
+        let vec_opt = self.memory.get(&key);
+        match vec_opt {
+            None => None,
+            Some(vect) => Some(&vect[..]),
+        }
+    }
+
+    fn del(&mut self, key: State) {
+        unimplemented![];
+    }
+}
+
+fn volatile_database<const N: usize, G>(game: &G) -> Result<TestDB>
+where
+    G: Extensive<N> + Game,
+{
+    let db = TestDB::initialize();
+    Ok(db)
+}
 
 #[cfg(test)]
-mod test {
+mod tests {
+    use crate::database::{KVStore, Tabular};
+    use crate::game::mock;
+    use crate::game::{
+        Bounded, ClassicPuzzle, DTransition, Extensive, Game, GameData,
+        SimpleSum,
+    };
+    use crate::interface::{IOMode, SolutionMode};
+    use crate::model::SimpleUtility;
+    use crate::model::{State, Turn};
+    use crate::node;
+    use crate::solver::record::surcc::RecordBuffer;
+    use anyhow::Result;
+    use std::collections::{HashMap, VecDeque};
+
+    use super::{
+        discover_child_counts, reverse_bfs_solver, volatile_database, TestDB,
+    };
+
+    struct GameNode {
```

Comment:
This whole graph construct should also be replaced with `game::mock::Session`. That being said, this is also repeated code between this test module and the one in the cyclic solver. If you ever want to make test utilities, they can go in the `module/test.rs` file, which is for testing utilities (in this case, `solver/test.rs`).